components/omega/doc/design/Decomp.md

```diff
@@ -0,0 +1,286 @@
+
+In the future, it may be desireable to perform parts of the calculation
+(eg the communication-dominated barotropic mode) on a smaller 
+partition. We may need to support multiple partitions of the
+same mesh on different numbers of MPI ranks.
+
+### 2.12 Desired: Multiple domains, sub-blocking or coloring
+
+For some future capabilities (eg local timestepping in regionally
+focused mesh locations), it may be desireable to decompose the
+global domain into smaller subdomains and over-subscribe subdomains
+to a rank, combining low- and high-resolution subdomains for load
+balancing. It is unclear whether this is best achieved with a multiple
+sub-block structure (as in older MPAS) or accomodating this within a
+weighted partitioning. Identifying the mesh regions using a coloring
+of the mesh may be helpful.
+
+## 3 Algorithmic Formulation
+
+Most of the algorithms required are embedded and documented in the
+relevant packages (eg Metis multi-level k-way partitioning). 
+
+## 4 Design
+
+The design presented here is initially for decomposing the global
+domain into a single block per MPI rank and does not yet include
+weighted decompositions. However, it should not prevent these
+options from being added later. The parmetis package will be
+used initially to provide metis back compatibility while reducing
+memory use over a serial metis approach. The cell mesh indices and
+adjacency will be read from the non-decomposed graph.info files in
+metis format, allowing the adjacency info to be read serially
+before the parallel IO is set up.
+
+Once the primary (cell) mesh is partitioned, the edge and vertex
+index spaces will be assigned based on the adjacency to the local
+cells, as in the current MPAS model. Because much of this information
+will be later replicated in a mesh class with YAKL arrays, this
+decomposition structure will be destroyed after initialization and
+any related setup of halos and other infrastructure.
+
+### 4.1 Data types and parameters
+
+#### 4.1.1 Parameters 
+
+The decomposition includeds a public enum to define the
+supported partitioning method. Initially, this will support
+the default Metis method, but others can be added:
+
+```c++
+    enum partMethod {
+       partMethodUnknown,        ///< undefined for error checking
+       partMethodMetisKWay,      ///< default KWay method in metis
+       }
+```
+
+Other methods, like the Metis GeomKWay (combination of space-filling
+curve with KWay) or Zoltan options may be added later.
+
+Eventually, an enum defining options for computing weights for
+weighted partitions may also be required, but will not appear in
+this initial implementation.
+
+Another parameter is the `haloWidth` that defines the width
+of the halo needed to provide neighbor information for all
+local operators. The minimum is typically 3 for current
+algorithms. 
+
+These parameters and the mesh input file will be read as part of
+the input configuration file in a decomp configuration group:
+
+```yakl
```

Comment:
We should fix this somewhere.  It should be `yaml`, not `yakl`. I'm sure this is the last time this will ever happen to us :-p